packages/app/src/sandbox/eval/transpilers/svelte/svelte-worker.js

```diff
@@ -1,21 +1,65 @@
+import semver from 'semver';
 import { buildWorkerError } from '../utils/worker-error-handler';
 import { buildWorkerWarning } from '../utils/worker-warning-handler';
 
 // Allow svelte to use btoa
 self.window = self;
 
-self.importScripts(['https://unpkg.com/svelte@^2.0.0/compiler/svelte.js']);
-
 self.postMessage('ready');
 
-declare var svelte: {
-  compile: (code: string, options: Object) => { code: string },
-};
+// declare var svelte: {
+//   compile: (code: string, options: Object) => { code: string },
+// };
+
+function getV3Code(code, path) {
+  self.importScripts(['https://unpkg.com/svelte@3.1.0/compiler.js']);
+  try {
+    const { js, warnings } = self.svelte.compile(code, {
```

Comment:
could we consider something like this?

```js
self.importScripts(['https://unpkg.com/cosmiconfig@5.2.0']);
try {
  const explorer = cosmiconfig('svelte');
  const {config} = await explorer.search('svelte')
  if (config) code = self.svelte.preprocess(code, config.preprocess);

  const { js, warnings } = self.svelte.compile(code, {
    // ...
```

user-side code:

_svelte.config.js_
```
const ts = require('svelte-ts-preprocess')
const sass = require('svelte-preprocess-sass')

module.exports = {
  preprocess: [ts, sass, require('./custom-preprocessor')]
}
```
I'm not sure that `cosmiconfig` will work in the browser, and maybe there are better approaches to load the config with preprocessors for codesandobx. But this kind of addition will help users set their custom svelte preprocessors (like https://github.com/ls-age/svelte-preprocess-sass or https://github.com/PaulMaly/svelte-ts-preprocess)

Reply:
Cosmic config doesn't seem to work in the browser :/
https://codesandbox.io/s/nnn2lmnkqm

This is an option we will do but I think we will need to check for the file ourselfs and get the data out

I am also not sure how these preprocessors will work in the browser. We have our own preprocessors for ts and sass and it would be great if we could use them :(

Reply:
It miiiight maybe work in the worker, we mock `fs` there and some other utils. But there is also a valid chance that it won't work. Especially since I do this 😅 : https://github.com/CompuIves/codesandbox-client/blob/b75e9969baa73478b05f49e7d8da85b08945f5ad/packages/app/config/polyfills.js#L14

Reply:
I think it makes sense to keep the simple option (maybe add some hardcoded support for sass/others) in the browser and give the option to move to a container if there is advanced configuration needed.

Reply:
maybe, besides the hardcoded support for sass and other preprocessors, it would be useful to hardcode something like `require('/svelte.config.js')` too for custom configs? to provide an option to also add some custom preprocessors by a user. And that would be the user's responsibility, how that preprocessors work in the browser